### PR TITLE
Build SDL2 libraries in CI

### PR DESCRIPTION
**Pull request by Beyley**
# Summary of the PR
Builds SDL2 from source using the CI

# Related issues, Discord discussions, or proposals
https://discord.com/channels/521092042781229087/587346162802229298/1102410331524956171

### CODE REVIEW

**Comment by Perksey** [comment]
CI is passing, can we merge this @Beyley?

**Comment by Beyley** [comment]
> CI is passing, can we merge this @Beyley?

Ready for review/merge!

**Review by Perksey — COMMENTED** [review]
Just one Q and this may be good to merge.

Also can you fix the duplicate PackageReference in Maths? appreciate this has nothing to do with you, guess is that dependabot went wonky.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -26,9 +26,12 @@ jobs:
...
     runs-on: ${{ github.repository == 'dotnet/Silk.NET' && startsWith(github.ref, 'refs/tags/') && 'windows-quick' || github.repository == 'dotnet/Silk.NET' && 'windows-2022' || 'windows-latest' }}
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3.5.2
       with:
-        submodules: 'true'
+        submodules: 'false'
+    - name: Checkout submodules, configure git
+      run: |
+        git -c submodule.third_party/git-hooks.update=none submodule update --init --recursive
```
[the remark this answers is not in the document]
Comment (Beyley) [reply .github/workflows/build.yml:34]: Ah yeah i was just testing things trying to get the self-hosted runner to stop choking, i can revert the relevant commits to this file

**Comment by Beyley** [comment]
> Just one Q and this may be good to merge.
> 
> Also can you fix the duplicate PackageReference in Maths? appreciate this has nothing to do with you, guess is that dependabot went wonky.

I checked the csproj file and dont see a duplicate reference, are you sure its in *this* branch, or is it something in `main`?

**Comment by Perksey** [comment]
Oh... It's green now

**Review by Perksey — APPROVED** [review]

**Pull request merged**